api/app/registration/api.py

```diff
@@ -1,21 +1,175 @@
-from sqlalchemy.exc import SQLAlchemyError
-from app.utils import errors
-from app import LOGGER
 from app.registration.models import RegistrationSection
 from app.registration.models import RegistrationQuestion
 from app.registration.models import RegistrationForm
-from app.registration.models import Offer
 from app.registration.mixins import RegistrationFormMixin, RegistrationSectionMixin, RegistrationQuestionMixin
+from app.utils.auth import verify_token
+import traceback
+from flask import g, request
+from flask_restful import  fields, marshal_with
+from sqlalchemy.exc import SQLAlchemyError
 from app.events.models import Event
+from app.registration.models import Offer
+from app.registration.mixins import RegistrationMixin
 from app.users.models import AppUser
-from app import db
-from app.utils.auth import auth_required, admin_required, verify_token
-from flask import g, request
+from app import db, LOGGER
+from app.utils import errors
+from app.utils.auth import auth_required, admin_required
+
+OFFER_EMAIL_BODY = """
+Dear {} {} {},
+
+Congratulations on you offer
+
+Offer Details \n
+user_id {} \n
+event_id {} \n
+offer_date {} \n
+expiry_date {} \n
+payment_required {} \n
+travel_award {} \n
+accommodation_award {} \n
+
+Kind Regards,
+The Baobab Team
+"""
+
+def offer_info(offerEntity):
+    return {
+        'user_id': offerEntity.user_id,
+        'event_id': offerEntity.event_id,
+        'offer_date': offerEntity.offer_date,
+        'expiry_date': offerEntity.expiry_date,
+        'payment_required': offerEntity.payment_required,
+        'travel_award': offerEntity.travel_award,
+        'accommodation_award': offerEntity.accommodation_award
+    }
+
+class OfferAPI(RegistrationMixin, restful.Resource):
+    put_offer_fields = {
+        'user_id': fields.Integer,
+        'accepted': fields.Boolean,
+        'rejected': fields.Boolean,
+        'rejected_reason': fields.String
+    }
+
+    @auth_required
+    @marshal_with(put_offer_fields)
+    def put(self):
+        # update existing offer
+        args = self.req_parser.parse_args()
+        offer_id = args['id']
+        accepted = args['accepted']
+        rejected = args['rejected']
+        rejected_reason = args['rejected_reason']
+
+        offer = db.session.query(Offer).filter(Offer.id == offer_id).first()
+
+        if not offer:
+            return errors.OFFER_NOT_FOUND
+        else:
+            return
+
+        offerValues = Offer(
+            accepted = accepted,
+            rejected = rejected,
+            rejected_reason = rejected_reason
+            
+        )
+     
+        try:
+            offer = db.session.query(Offer).filter(Offer.id == offer_id).first()
+            offer.updated_at = datetime.now()
+            user_id = verify_token(request.headers.get('Authorization'))['id']
+
+            if offer and (not offer.user_id == user_id):
+                return errors.FORBIDDEN
+            else:
+                db.session.add(offerValues)
```

Comment:
I'm a bit confused here - it seems like you're adding a new offer record with the accepted, rejected and rejected_reason values rather than updating the existing offer (Which is what should happen) ?